src/Features/Core/Portable/ExternalAccess/VSTypeScript/Api/VSTypeScriptDocumentNavigationServiceWrapper.cs

```diff
@@ -6,29 +6,43 @@
 using System.Threading;
 using Microsoft.CodeAnalysis.Navigation;
 using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.Shared.Utilities;
 
 namespace Microsoft.CodeAnalysis.ExternalAccess.VSTypeScript.Api
 {
     internal readonly struct VSTypeScriptDocumentNavigationServiceWrapper
     {
         private readonly IDocumentNavigationService _underlyingObject;
+        private readonly IWorkspaceThreadingServiceProvider _threadingProvider;
 
-        public VSTypeScriptDocumentNavigationServiceWrapper(IDocumentNavigationService underlyingObject)
-            => _underlyingObject = underlyingObject;
+        public VSTypeScriptDocumentNavigationServiceWrapper(
+            IDocumentNavigationService underlyingObject,
+            IWorkspaceThreadingServiceProvider threadingProvider)
+        {
+            _underlyingObject = underlyingObject;
+            _threadingProvider = threadingProvider;
+        }
 
         public static VSTypeScriptDocumentNavigationServiceWrapper Create(Workspace workspace)
-            => new(workspace.Services.GetRequiredService<IDocumentNavigationService>());
+            => new(workspace.Services.GetRequiredService<IDocumentNavigationService>(),
+                   workspace.Services.GetRequiredService<IWorkspaceThreadingServiceProvider>());
 
         [Obsolete("Call overload that takes a CancellationToken", error: false)]
         public bool TryNavigateToPosition(Workspace workspace, DocumentId documentId, int position, int virtualSpace = 0, OptionSet? options = null)
             => this.TryNavigateToPosition(workspace, documentId, position, virtualSpace, options, CancellationToken.None);
 
         [Obsolete("Call overload that doesn't take options", error: false)]
         public bool TryNavigateToPosition(Workspace workspace, DocumentId documentId, int position, int virtualSpace, OptionSet? options, CancellationToken cancellationToken)
-            => _underlyingObject.TryNavigateToPosition(workspace, documentId, position, virtualSpace, NavigationOptions.Default, cancellationToken);
+        {
+            var obj = _underlyingObject;
+            return _threadingProvider.Service.Run(() => obj.TryNavigateToPositionAsync(workspace, documentId, position, virtualSpace, NavigationOptions.Default, cancellationToken));
+        }
 
-        /// <inheritdoc cref="IDocumentNavigationService.TryNavigateToPosition"/>
+        /// <inheritdoc cref="IDocumentNavigationService.TryNavigateToPositionAsync"/>
         public bool TryNavigateToPosition(Workspace workspace, DocumentId documentId, int position, int virtualSpace, CancellationToken cancellationToken)
-            => _underlyingObject.TryNavigateToPosition(workspace, documentId, position, virtualSpace, NavigationOptions.Default, cancellationToken);
+        {
+            var obj = _underlyingObject;
```

Comment:
Why the local?

Reply:
for some reason this type is a struct... and lambdas in structs can't capture instance members of 'this' :(